src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -55,6 +55,7 @@ public static partial class RequestDelegateFactory
         private static readonly MemberExpression HttpResponseExpr = Expression.Property(HttpContextExpr, nameof(HttpContext.Response));
         private static readonly MemberExpression RequestAbortedExpr = Expression.Property(HttpContextExpr, nameof(HttpContext.RequestAborted));
         private static readonly MemberExpression UserExpr = Expression.Property(HttpContextExpr, nameof(HttpContext.User));
+        private static readonly MemberExpression FormFileExpr = Expression.Property(HttpContextExpr, nameof(HttpContext.Request.Form.Files));
```

Comment:
This isn't how we would do this. Reading the form body should be done asynchronously. We'd probably store state on the FactoryContext that describes this is a form body that needs files. Take a look at [BindParameterFromBody](https://github.com/dotnet/aspnetcore/blob/34155282df177259d9b928080cbda9ca15a3afb3/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L715) and [HandleRequestBodyAndCompileRequestDelegate](https://github.com/dotnet/aspnetcore/blob/34155282df177259d9b928080cbda9ca15a3afb3/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L473)